### PR TITLE
Update webplatform.yaml to account for bug 1594717.

### PR DESCRIPTION
**Pull request by emilio**
In https://bugzilla.mozilla.org/show_bug.cgi?id=1594717 a fair amount of
components were changed.

### CODE REVIEW

File: src/data/webplatform.yaml [hunk 1]
```diff
@@ -11,33 +11,36 @@ products:
...
+  'DOM: UI Events & Focus Handling',
   'DOM: Web Authentication',
   'DOM: Web Crypto',
   'DOM: Web Payments',
-  'DOM: Web Storage',
+  'DOM: Window and Location',
   'DOM: Workers',
-  'Drag and Drop',
+  'DOM: postMessage',
   'Editor',
```
Comment (annevk) [inline src/data/webplatform.yaml]: DOM: Editor

Comment (emilio) [reply src/data/webplatform.yaml]: Gah, missed that one, thanks!

File: src/data/webplatform.yaml [hunk 2]
```diff
@@ -11,33 +11,36 @@ products:
...
   'DOM: CSS Object Model',
   'DOM: Device Interfaces',
+  'DOM: Drag & Drop',
+  'DOM: Editor',
   'DOM: Events',
-  'DOM: IndexedDB',
   'DOM: File',
+  'DOM: Forms',
+  'DOM: HTML Parser',
+  'DOM: Mozilla Extensions',
```
Comment (fiennyangeln) [inline src/data/webplatform.yaml]: I did a quick check on the advance search tab of bugzilla [website](https://bugzilla.mozilla.org/buglist.cgi?resolution=---&query_format=advanced&component=DOM%3A%20Mozilla%20Extensions&product=Core) and I wonder does this component exists?

Comment (emilio) [reply src/data/webplatform.yaml]: It was just moving it around to be alphabetically sorted, but I guess this component no longer exists (before the bug mentioned there?)

Comment (emilio) [reply src/data/webplatform.yaml]: Removed it in a separete commit.

**Comment by emilio** [comment]
I cannot merge this on my own, so help with that would be appreciated :)

**Comment by emilio** [comment]
Thanks!

**Pull request merged**